pallets/capacity/src/migration/v3.rs

```diff
@@ -0,0 +1,159 @@
+use crate::{BalanceOf, BlockNumberFor, Config, FreezeReason, Pallet, StakingAccountLedger};
+use frame_support::{
+	pallet_prelude::{GetStorageVersion, IsType, Weight},
+	traits::{
+		fungible::MutateFreeze, Get, LockIdentifier, LockableCurrency, OnRuntimeUpgrade,
+		StorageVersion,
+	},
+};
+use parity_scale_codec::Encode;
+use sp_core::hexdisplay::HexDisplay;
+use sp_runtime::Saturating;
+
+const LOG_TARGET: &str = "runtime::capacity";
+
+#[cfg(feature = "try-runtime")]
+use sp_std::vec::Vec;
+
+const STAKING_ID: LockIdentifier = *b"netstkng";
+
+/// The OnRuntimeUpgrade implementation for this storage migration
+pub struct MigrationToV3<T, OldCurrency>(sp_std::marker::PhantomData<(T, OldCurrency)>);
+impl<T, OldCurrency> MigrationToV3<T, OldCurrency>
+where
+	T: Config,
+	OldCurrency: 'static + LockableCurrency<T::AccountId, Moment = BlockNumberFor<T>>,
+	OldCurrency::Balance: IsType<BalanceOf<T>>,
+{
+	/// Translate capacity staked locked deposit to frozen deposit
+	pub fn translate_lock_to_freeze(account_id: T::AccountId, amount: OldCurrency::Balance) {
+		OldCurrency::remove_lock(STAKING_ID, &account_id); // 1r + 1w
+		T::Currency::set_freeze(&FreezeReason::Staked.into(), &account_id, amount.into())
+			.unwrap_or_else(|err| {
+				log::error!(target: LOG_TARGET, "Failed to freeze {:?} from account 0x{:?}, reason: {:?}", amount, HexDisplay::from(&account_id.encode()), err);
+			}); // 1w
+		log::info!(target: LOG_TARGET, "🔄 migrated account 0x{:?}, amount:{:?}", HexDisplay::from(&account_id.encode()), amount.into());
+	}
+}
+
+impl<T: Config, OldCurrency> OnRuntimeUpgrade for MigrationToV3<T, OldCurrency>
+where
+	T: Config,
+	OldCurrency: 'static + LockableCurrency<T::AccountId, Moment = BlockNumberFor<T>>,
+	OldCurrency::Balance: IsType<BalanceOf<T>>,
+{
+	fn on_runtime_upgrade() -> Weight {
+		let on_chain_version = Pallet::<T>::on_chain_storage_version(); // 1r
+
+		if on_chain_version.lt(&3) {
+			log::info!(target: LOG_TARGET, "🔄 Capacity Locks->Freezes migration started");
+			let mut maybe_count = 0u32;
+			StakingAccountLedger::<T>::iter()
+				.map(|(account_id, staking_details)| (account_id, staking_details.active))
+				.for_each(|(staker, mut amount)| {
```

Comment:
nit: Having a `mut` variable in foreach looks very confusing. Why not just define a new variable like below?

`let total_freeze = amount.saturating_add(Pallet::<T>::get_unlocking_total_for(&staker));`